torch/_dynamo/eval_frame.py

```diff
@@ -643,6 +645,177 @@ def __bool__(self):
         )
 
 
+class _AddRuntimeAssertsInInputConstraint(torch.fx.interpreter.Transformer):
```

Comment:
Hmmph, I still am failing to get the big picture here.

Let's suppose that I trace and export a model that does if x.size(3) > 10 do something else do something else. Furthermore, let's suppose I didn't pass in any explicit constraints (which I assume means "try to make everything as dynamic as possible"). Then you will end up with a model which requires x.size(3) > 10 to be sound. But you aren't going to have a constraint for it (because it's dynamic by default). Do you still want to put the tests into the graph?

More generally, it seems to me that you are trying to solve the problem of "what to do with guards" by directly putting them in the graph. But does this make sense? Why is this not tested out of band?

Reply:
My thinking is that Avik's surfacing of guards will output those more fine-grained constraints directly to user (e.g x.size(3) > 10). And we suggest user to use those constraints directly on the input to the export call so that the exported module is more sound. And then, my pass will translate the new constraints into runtime assertions. 

One thing we can explore is, we can put the constraints derived from guards directly into graph after we finish exporting (not asking user to pass in the constraint).

Reply:
I guess, my point is, it's inconsistent to only do directly given constraints, and not also do implicit constraints from guards.

Reply:
Yeah I agree. It should also convert the implicit constraints as well. Can do it in the follow up diff.

Reply:
This is worth doing now, and is more important, and the primary reason for my requested changes.